cluster/gke/util.sh

```diff
@@ -370,13 +370,27 @@ function test-teardown() {
   # instances, but we can safely delete the cluster before the firewall.
   #
   # NOTE: Keep in sync with names above in test-setup.
-  "${GCLOUD}" compute firewall-rules delete "${CLUSTER_NAME}-http-alt" \
-    --project="${PROJECT}" &
-  "${GCLOUD}" compute firewall-rules delete "${CLUSTER_NAME}-nodeports" \
-    --project="${PROJECT}" &
+  for fw in "${CLUSTER_NAME}-http-alt" "${CLUSTER_NAME}-nodeports" "${FIREWALL_SSH}"; do
+    if [[ -n $("${GCLOUD}" compute firewall-rules --project "${PROJECT}" describe "${fw}" --format='value(name)' 2>/dev/null || true) ]]; then
+      "${GCLOUD}" compute firewall-rules delete "${fw}" --project="${PROJECT}" --quiet &
+    fi
+  done
 
   # Wait for firewall rule teardown.
   kube::util::wait-for-jobs || true
+
+  # It's unfortunate that the $FIREWALL_SSH rule and network are created in
+  # kube-up, but we can only really delete them in test-teardown. So much for
+  # symmetry.
```

Comment:
I just parallelized them there. We actually don't want to delete the firewall rules until after the `container clusters delete`, but I can't tell why we don't just delete them in `kube-down` after the call to `container clusters delete`, except that they're test-only. I suppose this is mostly irrelevant, though, because the GKE provider is really a synthetic provider for test.